crates/core/src/db/datastore/traits.rs

```diff
@@ -378,14 +378,20 @@ pub trait Tx {
     type TxId;
 
     fn begin_tx(&self) -> Self::TxId;
-    fn release_tx(&self, tx: Self::TxId);
+    fn release_tx(&self, ctx: &ExecutionContext, tx: Self::TxId);
 }
 
 pub trait MutTx {
     type MutTxId;
 
     fn begin_mut_tx(&self) -> Self::MutTxId;
+    fn rollback_mut_tx_with_context(&self, ctx: &ExecutionContext, tx: Self::MutTxId);
```

Comment:
Should this just stay `rollback_mut_tx`, and the test-only `rollback_mut_tx` get some worse name? Lots of things are going to plump around an ExecutionContext to the point that it doesn't seem exceptional or worth calling out.

Reply:
Yeah. Like I said, I'm happy to stamp another PR that changes em later if that's easier.